src/electron/platform/android/unified-result-builder.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { UnifiedScanCompletedPayload } from 'background/actions/action-payloads';
+import { generateUID, UUIDGeneratorType } from 'common/uid-generator';
+import { ApplicationPropertiesDelegate } from 'electron/common/application-properties-provider';
+import { ScanResults } from 'electron/platform/android/scan-results';
+import {
+    convertScanResultsToUnifiedResults,
+    ConvertScanResultsToUnifiedResultsDelegate,
+} from 'electron/platform/android/scan-results-to-unified-results';
+import {
+    convertScanResultsToUnifiedRules,
+    ConvertScanResultsToUnifiedRulesDelegate,
+} from 'electron/platform/android/scan-results-to-unified-rules';
+
+export type UnifiedScanCompletedPayloadBuilder = (scanResults: ScanResults) => UnifiedScanCompletedPayload;
+
+export const createBuilder = (
+    getUnifiedResults: ConvertScanResultsToUnifiedResultsDelegate,
+    getUnifiedRules: ConvertScanResultsToUnifiedRulesDelegate,
+    uuidGenerator: UUIDGeneratorType,
+    applicationPropertiesDelegate: ApplicationPropertiesDelegate,
+) => (scanResults: ScanResults): UnifiedScanCompletedPayload => {
+    const payload: UnifiedScanCompletedPayload = {
+        scanResult: getUnifiedResults(scanResults, uuidGenerator),
+        rules: getUnifiedRules(scanResults, uuidGenerator),
+        toolInfo: {
+            scanEngineProperties: {
+                name: 'axe-android',
+                version: scanResults.axeVersion,
+            },
+            applicationProperties: applicationPropertiesDelegate(),
+        },
```

Comment:
Is there a place where we might want to separate the tool info generation out to? I see that we use `EnvironmentInfoProvider` to generate similar properties for the `axe-core` results--is there a particular reason we would rather encapsulate that logic in here for android vs following a similar pattern to the 'axe-core' info? Definitely a minor thing, but it just feels a little off to me that we're hardcoding `name: 'axe-android'` into this file, which doesn't seem like it should know/care about the name of the scanning engine in my opinion.

Reply:
I implement the change in this PR... I was a litte hesitant to do it because we need the `scanResults` object to get the `axeVersion` value but on comment from review seems enough to just do it.